### DISCUSSION TITLE
Issue with SMPClientReadOnly URL Provider in Phase4Standalone App Using Local smp-query-webapp

### DISCUSSION THREAD

**Discussion by mohamedamineboughrara** [phax/phase4#286] [General]
I'm currently using the Phase4Standalone application to send messages through the smp-query-webapp running on my localhost. In my PeppolSenderController, I am trying to set the URL provider on the SMPClientReadOnly to use the smp.url configured in my application.properties.

I’ve encountered the following issues:

When I attempt to set the URL provider with the smp.url from application.properties, I receive an error indicating SMPHost must not be null.
Even after configuring the URL provider manually within the controller (using the smp.url), the application still throws the SMPHost must not be null error.
My goal is to make the Phase4Standalone app able to search through the instance of smp-query-webapp running on localhost. I’d like to know if there’s any workaround or recommended way to make Phase4Standalone correctly interact with the local smp-query-webapp for SMP lookups.

Any advice or guidance on how to configure this setup would be greatly appreciated!
Thank you

**Comment by phax** [comment]
Hi @mohamedamineboughrara,
If you are using phase4-peppol-standalone you don't need the smp-query-webapp. Just use the built-in SMP client - that's way more efficient. The class `SMPClientReadOnly` (or `SMPClient`) does exactly what the smp-query-webapp does.
The smp-query-webapp is only for the non-Java guys that want an easy way to perform a secure SMP lookup. There is no predefined way to include those two.

The `smp.url` configuration property is only needed to check inbound messages, to check if they are registered on **your SMP** and are therefore legitimate a specific request.
